finat/quadrature.py

```diff
@@ -1,150 +1,196 @@
 from __future__ import absolute_import, print_function, division
+from six import with_metaclass
 
-import numpy as np
-from .gauss_jacobi import gauss_jacobi_rule
-from .points import StroudPointSet, PointSet, TensorPointSet, GaussLobattoPointSet
-import FIAT
-import gem
+from abc import ABCMeta, abstractproperty
+from functools import reduce
 
+import numpy
 
-def make_quadrature(cell, degree, preferred_quadrature):
-    '''Return a quadrature rule of the suggested degree in accordance with
-    the quadrature preferences provided.'''
+import gem
+from gem.utils import cached_property
 
-    # Currently just do the dumb thing. Smart quadrature happens later.
-    return CollapsedGaussJacobiQuadrature(cell, degree)
+from FIAT.reference_element import QUADRILATERAL, TENSORPRODUCT
+# from FIAT.quadrature import compute_gauss_jacobi_rule as gauss_jacobi_rule
+from FIAT.quadrature_schemes import create_quadrature as fiat_scheme
 
+from finat.point_set import PointSet, TensorPointSet
 
-class QuadratureRule(object):
-    """Object representing a quadrature rule as a set of points
-    and a set of weights.
 
-    :param cell: The :class:`~FIAT.reference_element.ReferenceElement`
-    on which this quadrature rule is defined.
+def make_quadrature(ref_el, degree, scheme="default"):
```

Comment:
This mostly mimics `FIAT.quadrature_schemes.create_quadrature`. Preferred quadrature argument currently removed. I argue for a gradual approach where we first have sum factorisation with the potential need to manually specify the quadrature rule, and then we can improve the automatic quadrature scheme selection.